markdown_lab/network/client.py

```
@@ -201,6 +206,143 @@ def _request_with_retries(
+        # Create semaphore to limit concurrent requests
+        max_concurrent = min(self.config.max_concurrent_requests, len(urls))
+        semaphore = asyncio.Semaphore(max_concurrent)
+
+        async def fetch_one(url: str) -> tuple[str, Optional[str]]:
+            async with semaphore:
+                try:
+                    # Add throttling delay
+                    await asyncio.sleep(1.0 / self.config.requests_per_second)
+                    content = await self.get_async(url, **kwargs)
+                    return url, content
+                except Exception as e:
```

Pylint (3.3.7)

[warning] Catching too general exception Exception

(W0718)